x/auth/keeper/keeper.go

```diff
@@ -317,3 +338,18 @@ func (ak AccountKeeper) NonAtomicMsgsExec(ctx context.Context, signer sdk.AccAdd
 
 	return msgResponses, nil
 }
+
+// MigrateAccountNumberUnsafe migrates auth's account number to accounts's account number
+// and delete store entry for auth legacy GlobalAccountNumberKey.
+//
+// Should only use in an upgrade handler for migrating account number.
+func (ak AccountKeeper) MigrateAccountNumberUnsafe(ctx context.Context) error {
```

Comment:
personal preference: this could be defined as static method in the package instead that receive an account keeper instance. Both ways the method is public but it would not be exposed by the keeper instance.